sw/device/sca/otbn_vertical/ecc256_keygen_serial.c

```diff
@@ -234,6 +270,90 @@ void ecc256_ecdsa_secret_keygen_batch(const uint8_t *data, size_t data_len) {
                             kEcc256SeedNumWords * 4);
 }
 
+void add_arrays(uint8_t *dest, uint8_t *source, size_t dest_len,
+                size_t source_len) {
+  uint16_t temp = 0;
+
+  for (size_t i = 0; i < source_len; i++) {
+    temp += (uint16_t)source[i] + dest[i];
+    dest[i] = (uint8_t)(temp & 0x00FF);
+    temp >>= 8;
+  }
+
+  for (size_t i = source_len; i < dest_len; i++) {
+    temp += (uint16_t)dest[i];
+    dest[i] = (uint8_t)(temp & 0x00FF);
+    temp >>= 8;
+  }
```

Comment:
The way we use this function, overflow can happen with only 2^(-64) probability.
It may be useful to add the overflow check if we end up using this function elsewhere but I'd rather leave it for when it happens. For now I've added a comment saying that the user should ensure that `dest_len` is enough to store the result without overflowing.

Reply:
Thanks @vrozic for adding the comment and explaining here. This makes sense to me.

Questions:
1. Does the overflow even matter in our scenario?
2. How do we handle this on the capture side? Are the overflows handled there, i.e., an overflow on the target leads to a digest mismatch?

Reply:
1. I haven't tested this case but i believe that the capture would break in the unlikely case that this overflow happens.
2. On the capture side we ignore the possibility of the overflow. This means that `seed_fixed` would end up being one byte longer than expected. Most likely we would get some error when transmitting data. One way to address this would be to do something like `seed_fixed_int %= 2 ** seed_len` on the capture side which would make the overflow not matter. However I don't think it's important to address this because the possibility is only theoretical.

Reply:
Thanks for providing more insight @vrozic ! In this case it makes even more sense to leave things as you proposed.